Newer ghostscript has new enumeration names for the error codes.

--- libspectre/spectre-gs.c	2010-04-18 18:01:05.000000000 +0200
+++ libspectre/spectre-gs.c	2015-10-15 04:26:50.142190302 +0200
@@ -43,12 +43,12 @@
 	
 	if (code <= -100) {
 		switch (code) {
-			case e_Fatal:
+			case gs_error_Fatal:
 				fprintf (stderr, "fatal internal error %d", code);
 				return TRUE;
 				break;
 
-			case e_ExecStackUnderflow:
+			case gs_error_execstackoverflow:
 				fprintf (stderr, "stack overflow %d", code);
 				return TRUE;
 				break;
@@ -109,9 +109,9 @@
 		set = _spectre_strdup_printf ("%d %d translate\n", -x, -y);
 		error = gsapi_run_string_continue (ghostscript_instance, set, strlen (set),
 						   0, &exit_code);
-		error = error == e_NeedInput ? 0 : error;
+		error = error == gs_error_NeedInput ? 0 : error;
 		free (set);
-		if (error != e_NeedInput && critic_error_code (error)) {
+		if (error != gs_error_NeedInput && critic_error_code (error)) {
 			fclose (fd);
 			return FALSE;
 		}
@@ -126,7 +126,7 @@
 		read = fread (buf, sizeof (char), to_read, fd);
 		error = gsapi_run_string_continue (ghostscript_instance,
 						   buf, read, 0, &exit_code);
-		error = error == e_NeedInput ? 0 : error;
+		error = error == gs_error_NeedInput ? 0 : error;
 		left -= read;
 	}
 	
